ShopLens/ShopLensForms/ShopLens.cs

```diff
@@ -134,26 +109,32 @@ private void CAPTURE_Click(object sender, EventArgs e)
         {
             if (live_video.Image != null)
             {
+                //This line of code causes trouble when trying to identify items multiple times.
                 var image = (Image)live_video.Image.Clone();
-                capture_picture.Image = image;
+
+                //This line of code also causes trouble! Uncomment at your own risk!
+                //capture_picture.Image = image;
 
                 var ms = new MemoryStream();
                 image.Save(ms, System.Drawing.Imaging.ImageFormat.Jpeg);
 
                 var classificationResults = _imageClassifying.ClassifyImage(ms.ToArray());
 
-                var resultStrings = classificationResults.Select(pair => $"{pair.Key} - {(int)(pair.Value * 100)} percent.");
                 _textVoicer.SayMessage("This is");
 
                 //Order by probability values and take the first label name.
-                string mostConfidentResult = classificationResults.OrderByDescending(x => x.Value).FirstOrDefault().ToString();
-
-                //Format string so only the name of the product is said.
-                string[] results = mostConfidentResult.Split(',');
-
-                _textVoicer.SayMessage(results[0]);
-
-                Thread.Sleep(500);
+                string mostConfidentResult = classificationResults.OrderByDescending(x => x.Value).FirstOrDefault().Key;
+
+                if (mostConfidentResult == null) 
+                {
+                    _textVoicer.SayMessage(noLblError);
+                }
+                else
+                {
+                    _textVoicer.SayMessage(mostConfidentResult);
+                }
+
+                Thread.Sleep(500);  //Why is this here?
```

Comment:
Why is this here? Why do we need to stop the main Thread? What happens when you delete this?

Reply:
I noticed that if you CAPTURE too many times in a short span of time the program throws an exception related to the "Thread Bug".

Reply:
But this only happens when using the voice command "what is this" -  you can click the CAPTURE button as many times as you want.

Reply:
Good point, should delete this line